basics/plugin_structure/dynamic_plugins.md

```diff
@@ -70,15 +70,39 @@ When a plugin is being uninstalled or updated, the IDE waits synchronously for p
 
 ### Logging
 
-All events are tracked under `com.intellij.ide.plugins.DynamicPlugins` category in IDE log file.
+All events are tracked under `com.intellij.ide.plugins.DynamicPlugins` category in IDE log file. If a plugin fails to reload, the log will contain a cause as to why.
 
 ### Diagnosing Leaks
 
 To find leaks preventing clean unload, perform the following steps:
 
-- Set registry key `ide.plugins.snapshot.on.unload.fail` to `true`
-- Verify that IDE runs with VM parameter `-XX:+UnlockDiagnosticVMOptions`
-- Open the `.hprof` snapshot generated on plugin unload, look for the plugin ID string
-- Find the `PluginClassLoader` referencing the plugin ID string
-- Look at references to the `PluginClassLoader` instance.
-  Every one of them is a memory leak (or part of a memory leak) that needs to be resolved.
+1. Verify that the IDE is running with the VM parameter `-XX:+UnlockDiagnosticVMOptions`. If you are using the [Gradle](/tutorials/build_system/gradle_guide.md), the `runIde.jvmArgs += "-XX:+UnlockDiagnosticVMOptions"` otherwise you can [Configure JVM Options](https://www.jetbrains.com/help/idea/tuning-the-ide.html#procedure-jvm-options) to change them.
```

Comment:
Gradle is a [proper noun](https://en.wikipedia.org/wiki/Proper_and_common_nouns#Strong_and_weak_proper_names) and doesn't need an article. The second "the" should be a "then". I'd suggest: If you build your plugin using Gradle, then set runIde.jvmArgs += "-XX:+UnlockDiagnosticVMOptions". Otherwise, add this VM parameter in IntelliJ IDEA by configuring the JVM options.